### PR TITLE
Remove reentrancy library

### LINKED ISSUE #3217: Migrate reentrancy guard to sway-libs
In light of #10 being resolved (soon), migrate the reentrancy guard out of the stdlib and into https://github.com/FuelLabs/sway-libs since its use should not be standard.

### PR DESCRIPTION
**Pull request by jtriley-eth**
## Overview

Closes #3217, migrates the reentrancy library to Sway-libs.

> Notice: Awaiting merge and version bump from Sway-libs ([relevant pull request](https://github.com/FuelLabs/sway-libs/pull/70)).

## Changes

- remove reentrancy library
- remove reentrancy tests
- modifies documentation to point to new reentrancy guard url
- modifies imports of `std::reentrancy::` to `sway_libs::reentrancy`

### CODE REVIEW

**Comment by jtriley-eth** [comment]
Reentrancy library migrated. ([merged pr here](https://github.com/FuelLabs/sway-libs/pull/83))

File: docs/book/src/blockchain-development/calling_contracts.md [hunk 1]
```diff
@@ -78,12 +78,12 @@ fn main() {
 
 A common attack vector for smart contracts is [re-entrancy](https://docs.soliditylang.org/en/v0.8.4/security-considerations.html#re-entrancy). Similar to the EVM, the FuelVM allows for re-entrancy.
 
-A _stateless_ re-entrancy guard is included in the Sway standard library. The guard will panic (revert) at run time if re-entrancy is detected.
+A _stateless_ re-entrancy guard is included in the Sway-libs library. The guard will panic (revert) at run time if re-entrancy is detected.
```
Comment (mohammadfawaz) [inline docs/book/src/blockchain-development/calling_contracts.md:81]:
```suggestion
A _stateless_ re-entrancy guard is included in the [sway-libs](https://github.com/FuelLabs/sway-libs) library. The guard will panic (revert) at run time if re-entrancy is detected.
```

**Review by mohammadfawaz — COMMENTED** [review]
One small suggestion. Thanks!

**Review by mohammadfawaz — APPROVED** [review]

**Review by nfurfaro — APPROVED** [review]
LGTM.

File: docs/reference/src/code/operations/re_entrency/Forc.toml [hunk 2]
```diff
@@ -5,3 +5,4 @@ license = "Apache-2.0"
 name = "re_entrency"
 
 [dependencies]
+reentrancy = { git = "https://github.com/FuelLabs/sway-libs", tag = "v0.6.0" }
```
Comment (Braqzen) [inline docs/reference/src/code/operations/re_entrency/Forc.toml:8]:
```suggestion
reentrancy = { git = "https://github.com/FuelLabs/sway-libs", tag = "v0.6.0" }

```

**Review by Braqzen — APPROVED** [review]

**Pull request merged**